packages/experimental-package/src/components/RemoveModal/RemovalModal.js

```diff
@@ -0,0 +1,69 @@
+import React, { useState } from 'react';
+import { Modal, TextInput } from 'carbon-components-react';
+import PropTypes from 'prop-types';
+
+export const RemovalModal = ({
+  body,
+  className,
+  header,
+  onRequestClose,
+  onRequestSubmit,
+  open,
+  primaryButtonText,
+  resource,
+  secondaryButtonText,
+  subheader,
+  textConfirmation,
+}) => {
+  const [userInput, setUserInput] = useState('');
+  const onChangeHandler = (e) => {
+    setUserInput(e.target.value);
+  };
+  const primaryButtonDisabled = textConfirmation && userInput !== resource;
+  return (
+    <Modal
+      danger
+      className={`removal-modal ${className}`}
+      open={open}
+      modalHeading={header}
+      modalLabel={subheader}
+      primaryButtonText={primaryButtonText}
+      primaryButtonDisabled={primaryButtonDisabled}
+      secondaryButtonText={secondaryButtonText}
+      onRequestSubmit={onRequestSubmit}
+      onRequestClose={onRequestClose}>
+      <p className="removal-modal-body">{body}</p>
+      {textConfirmation && (
+        <TextInput
+          id="remove-confirmation"
```

Comment:
If an id is used in a component it should be unique so as not to clash with other ids that may be in the page. There should be a common mechanism for generating these ids for our components to use.

Reply:
Carbon components React has a shared common function like this: 
```
let lastId = 0;

export default function(prefix = 'id') {
  lastId++;
  return `${prefix}${lastId}`;
}
```
it tends to be used like this:
```
import uid from '../../somewhere-common/uniqueId';
...
this.uid = this.props.id || uid();
...
<input
  ...
  id={this.uid}
/>
<label htmlFor={this.uid} ... >
  ...
</label>
```

All their components use it when they need an id, instead of forcing the user to supply one. In this case we just need one because it's required by `<TextInput>`. It's of no value to the client of the component, and forcing them to supply it via a prop causes them needless extra work. 

We don't have a common function for generating ids for internal use by our components yet, but we probably should add one to the repo.

Reply:
good idea. i'll implement that functionality. would you prefer we implement our own id generating function or use a library like [uuid](https://www.npmjs.com/package/uuid)?

Reply:
In Carbon Vue we went with a an optional id. In effect it gets out of the way if the user specifies one. Can we do something like that?

https://github.com/carbon-design-system/carbon-components-vue/blob/master/packages/core/src/mixins/uid-mixin.js

Reply:
React uses the uniqueId in the new treeview component as follows which effectively allows the user to pass a value in, or get a generated value.

```
  const { current: id } = useRef(rest.id || uniqueId());
```